src/lib/execution.ts

```diff
@@ -115,6 +139,31 @@ export const generateExecutionId = (event: unknown) =>
     event,
   });
 
+// Used inside error path, do not throw
+export const sendFailureToDestinations = async (
+  failure: FailureResponse,
+  executionId: string
+): Promise<undefined | { error: ErrorWithContext }> => {
+  const errorDestinations = await loadExecutionErrorDestinations();
+
+  if ("error" in errorDestinations) {
```

Comment:
nit: similar to my comment above, it feels like it would be more natural to throw from within `loadExecutionErrorDestinations` instead (and move the `try` further up to cover that use case)